### ISSUE TITLE
How to interpret steps?

### ISSUE THREAD

**Issue by TDDFT** [tensorflow/ranking#47]
My eval data set contains 39540 queries, with list size 10 and batch size 128. Inside the data, there are 231820 distinct comparable pairs such that label_i>label_j. I did evaluation with 1 epoch. I use a data set generator like this:
`train_dataset.shuffle(5000).repeat(num_epochs).batch(_BATCH_SIZE)`

But for 1 epoch evaluation, I see 32475 global steps. This [stackoverflow](https://stackoverflow.com/questions/38340311/what-is-the-difference-between-steps-and-epochs-in-tensorflow) post says steps indicates batches. If so, for 1 epoch I should have 39540/128~309 steps, but the number of steps is way bigger than this and is of the same scale of number of queries.

Can someone give me a hint how steps are defined?

**Comment by xuanhuiwang** [comment]
I think "steps" indeed means "batches". In 1 step, 1 batch is feed. In your case, one eval will run about 309 steps.

I think there is a difference between number of steps for one eval vs the global steps. The global steps are usually for training. For example, at global step 32475 (note that this means a time point), one eval is started. This eval run one epoch with 309 steps and report the eval result. The eval result is plotted as the y axis and the x-axis is the global step 32475.

Let me know whether this makes sense.

**Comment by eggie5** [comment]
It depends on your Dataset that the input function returns. Most Datasets will have a `batch` step on the end so, one step to the dataset iterator will emit 1 batch. So, steps are batches and batches are queries.

**Comment by TDDFT** [comment]
I think I get it. Thanks!
